pkg/monitor/monitor.go

```diff
@@ -228,9 +227,26 @@ func (m *Monitor) driverMonitor() {
 				// For any Running pod on that node using volume by the driver, kill the pod
 				// Degraded nodes are not considered offline and pods are not deleted from them.
 				if node.Status == volume.NodeOffline || node.Status == volume.NodeDegraded {
+					// Only initialize the map when it is absolutely necessary
+					if len(k8sNodeNameToNodeMap) == 0 {
+						k8sNodeNameToNodeMap, err = m.getK8sNodeNameToNodeMap()
+						if err != nil {
+							log.Errorf("Error getting k8s node name to node map: %v", err)
+							continue
+						}
+					}
+					// Only initialize the map when it is absolutely necessary
+					if len(driverNodeToK8sNodeMap) == 0 {
+						driverNodeToK8sNodeMap = m.getVolumeDriverNodesToK8sNodeMap(nodes)
+					}
+					if _, ok := driverNodeToK8sNodeMap[node.StorageID]; !ok {
+						log.Errorf("Node with scheduler name %v and storage ID %v is not found in k8s cluster", node.SchedulerID, node.StorageID)
+						continue
+					}
+					k8sNode := driverNodeToK8sNodeMap[node.StorageID]
 					m.wg.Add(1)
 					// wait for 1 min if node is upgrading
-					go m.cleanupDriverNodePods(node)
+					go m.cleanupDriverNodePods(node, k8sNode, k8sNodeNameToNodeMap)
```

Comment:
I remember you mentioning that we are going to limit the number of go routines using the sched package. Is it being done in a separate PR?

Reply:
Yes, that is being handled as part of another ticket.